### PR TITLE
Password confirmation and firebase settings

### PR DESCRIPTION
**Pull request by umida-boltaeva**
I connected the app with the firebase. Users are showing up in authentication but not in the database.

### CODE REVIEW

File: src/firebase.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as firebase from 'firebase';
+
+const config = {
+    apiKey: process.env.REACT_APP_API_KEY,
```
Comment (nickmcmillan) [inline src/firebase.js:4]: You'll need to make sure that your `.env` file is in the root folder of the project, and that the variables names match up exactly, eg;
```
REACT_APP_API_KEY=abc123
```

I also found that when I was editing the `.env` file I had to stop and restart the dev server each time.

File: src/signup/signup.js [hunk 2]
```diff
@@ -9,16 +9,57 @@ import CssBaseline from "@material-ui/core/CssBaseline";
 import Typography from "@material-ui/core/Typography";
 import Button from "@material-ui/core/Button";
 import styles from "./styles";
-import { Form } from "reactstrap";
+
 const firebase = require("firebase");
```
Comment (nickmcmillan) [inline src/signup/signup.js]: You have a file at `src/firebase.js`. Inside this file is where you initialise firebase, and export it for use in the app.

So here in `signup.js` you should be importing `src/firebase`, not the firebase package itself. So I think this line should be replaced with 
```
import firebase from '../firebase'
```

File: src/signup/signup.js [hunk 3]
```diff
@@ -9,16 +9,57 @@ import CssBaseline from "@material-ui/core/CssBaseline";
...
   submitSignup = (e) => {
-    console.log("submitting!");
+    e.preventDefault();
+
+    if (!this.formIsValid()) {
+      this.setState({ signupError: "Passwords do not match!" });
+      return;
+    }
+
+    firebase
```
Comment (nickmcmillan) [inline src/signup/signup.js:52]: This is all really clear and easy to read 👍

File: src/signup/signup.js [hunk 4]
```diff
@@ -75,6 +117,16 @@ class SignupComponent extends React.Component {
               Submit
             </Button>
           </form>
+          {this.state.signupError ? (
```
Comment (nickmcmillan) [inline src/signup/signup.js]: Quick tip: you're doing an if/else statement here, but inside the else you have null. 
There's a better way to do this;

```
{this.state.signupError && (
  <Typography></Typography>
)}
```

You can learn about it here https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

File: src/signup/signup.js [hunk 5]
```diff
@@ -89,6 +141,25 @@ class SignupComponent extends React.Component {
       </main>
     );
   }
+
+  formIsValid = () => this.state.password === this.state.passwordConfirmation;
```
Comment (nickmcmillan) [inline src/signup/signup.js]: I'd suggest making the render() method last. Otherwise it's easy to miss these extra methods at the bottom of the file.

**Review by nickmcmillan — CHANGES_REQUESTED** [review]
So regarding the user data not being published into the database. 

You might need to double check you have created a database in firebase called `users` - and it might even help to manually add a document to the collection in the firebase dashboard.

Also take a look at my comments below. I'm hoping that with a few small changes you'll be pushing user data into your database in no time.

File: src/firebase.js (hunk 1 above)
Comment (umida-boltaeva) [reply src/firebase.js:4]: Yes, I checked that every variable name matches up exactly.

File: src/signup/signup.js (hunk 2 above)
Comment (umida-boltaeva) [reply src/signup/signup.js]: Ok, I replaced the line with `import firebase from '../firebase'`

File: src/signup/signup.js (hunk 4 above)
Comment (umida-boltaeva) [reply src/signup/signup.js]: So good to know, thank you! I read the documentation and changed the if/else statement.

File: src/signup/signup.js (hunk 5 above)
Comment (umida-boltaeva) [reply src/signup/signup.js]: Updated! Now the render() method is last.

**Pull request merged**